Account/FirefoxAccount.swift

```diff
@@ -349,6 +369,15 @@ open class FirefoxAccount {
         return client.destroyDevice(ownDeviceId: ownDeviceId, withSessionToken: session.sessionToken as NSData) >>> succeed
     }
 
+    fileprivate func registerOrUpdateDevice(session: TokenState, force: Bool = false) -> Success {
+        return FxADeviceRegistrator.registerOrUpdateDevice(self, sessionToken: session.sessionToken as NSData, force: force).bind { result in
+            if result.successValue != FxADeviceRegistrationResult.alreadyRegistered {
+                NotificationCenter.default.post(name: NotificationFirefoxAccountDeviceRegistrationUpdated, object: nil)
+            }
+            return succeed()
```

Comment:
You probably want `upon`, unless you specifically want to wait for the `NotificationCenter` work to be done before consumers get to act on the `Deferred` (which I think you don't, 'cos you discard the return value).

```
let d = FxADeviceRegistrator.registerOrUpdateDevice(…)
defer { return d }
d.upon { result in
    if result…
}
```